performance/performance_test.js

```diff
@@ -0,0 +1,46 @@
+const { spawn } = require('child_process');
+
+const numberOfRuns = 100;
+
+const getAverage = list => {
+  return list.reduce((res, value) => (res += value), 0) / list.length;
+};
+
+/**
+ * Calculates the average time taken of the given async test by running it multiple times.
+ * Each run is executed in a child process, to not share resources between test runs.
+ * The given async test must pass the time taken in a console log, for the performance test to work.
+ * @param {String} name - name of the test for the console log
+ * @param {String} testFilePath - absolute path to the test file
+ */
+const runAsyncTimeAverageTest = async (name, testFilePath) => {
+  const times = [];
+
+  for(let i = 0; i < numberOfRuns; i++) {
+    try {
+      const time = await new Promise((resolve, reject) => {
+        const proc = spawn("node", [testFilePath]);
```

Comment:
Candid question, do we really need to go throught a new child process every time ? Just calling a promise that returns the value was not enough ? I recall you talked about the more you run the test the more it took, I guess it's related to that ?

Reply:
maybe now that we don't do `Promise.all` it isn't needed anymore but I will need to test it out. This is safer because we know that each test is not impacting other runs (by sharing the same resources)

Reply:
But since you're doing a await they actually running one by one so it won't share anything right ?

Reply:
I tested it without the child processes, and the results changed in correlation to how many runs was performed. So it seems it is still needed

Reply:
If Rumesh or Jakub have an explaination for that I would be happy to know it

Reply:
TL;DR; Yes spawn new process will be faster than doing it in the same execution thread.

> The child_process.spawn() method spawns the child process asynchronously, without blocking the Node.js event loop
https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options
So first of all using `spawn` will not block the node event loop.

And NodeJS is single thread but using `spawn` let the capabilities to use ale the cores in the CPU

> a new process will be created and managed by your OS. That new process can be executed in parallel to the main process as long as your computer has at least 2 virtual CPU cores

https://stackoverflow.com/questions/33381796/does-node-child-process-spawn-run-on-multi-core-cpu

So yes `spawn` will create clean and new thread to execute this test file.

Reply:
spawning `node` for each test feels heavy. Maybe https://nodejs.org/api/vm.html can be useful and more lightweight, although I never used it personally.

Reply:
I looked into it and tried it, but I had some errors (`new vm.Script('node performance/get_time.js')` was throwing an error). As the waiting time for the performance tests to run is "not great, not terrible", and this PR isn't a priority, I will leave it as it is for now, because at least it works 🤷‍♀